SportHub/src/modules/AdminMainArticlesSection/components/AdminMainArticlesSection.js

```diff
@@ -0,0 +1,117 @@
+    }, [mainArticles])
+
+    const SaveChanges = () => {
+        let jsonPostRequest = []
+        for (let articleCount = 0; articleCount < mainArticles.length; articleCount++)
+            jsonPostRequest.push({
+                                OrderNumber: mainArticles[articleCount].order,
+                                ArticleId: mainArticles[articleCount].article.id,
+                                LanguageId: 1})
+        console.log(JSON.stringify(jsonPostRequest))
+        return JSON.stringify(jsonPostRequest)
+    }
+
+    const CancelChanges = () =>{
+        setMainArticles([])
+        setContent([])
+    }
+
+    useEffect(() =>{
+        if(typeof(setButtons) == "function"){
+            setButtons([{text: "Cancel", function: CancelChanges, isOutlined: true},
+                {text: "Save changes", function: SaveChanges, isOutlined: false}])
+        }
+    })
+
+    const GenerateAddMainArticleButton = () => {
+        if (mainArticles.length === 0)
+            return (
+                <div className={styles.add_one_more_article} onClick={() => AddNewMainArticle()}>
+                + Add main article
+                </div>
+            )
+    }
+
+    const AddNewMainArticle = () => {
+        if (mainArticles.length !== MAX_MAIN_ARTICLES_AMOUNT) {
+            setMainArticles([...mainArticles, {order: null,
+                                                    isTheLast: null,
+                                                    category: null,
+                                                    subcategory: null,
+                                                    team: null,
+                                                    article: null
+            }])
+        }
+    }
+
+    const DeleteMainArticle = (index) => {
+        const newList = [...mainArticles]
+        newList.splice(index, 1)
+        setMainArticles(newList);
+    }
+
+    const SaveData = (order, category, subcategory, team, article) => {
+        const newList = [...mainArticles]
+        newList[order].category = category
+        newList[order].subcategory = subcategory
+        newList[order].team = team
+        newList[order].article = article
+        setMainArticles(newList)
+    }
+
+    const GenerateArticlesForms = () => {
+        let contentList = []
+        setContent([])
+        for (let articleFormsCount = 0; articleFormsCount < mainArticles.length && articleFormsCount < MAX_MAIN_ARTICLES_AMOUNT; articleFormsCount++){
+            const newList = [...mainArticles];
+            newList[articleFormsCount].order = articleFormsCount
+            newList[articleFormsCount].isTheLast = articleFormsCount === newList.length - 1 && articleFormsCount !== MAX_MAIN_ARTICLES_AMOUNT - 1
+            setMainArticles(newList)
+
+            contentList[articleFormsCount] = (<AddMainArticle
+                            key={articleFormsCount}
+                            selectedOrder={mainArticles[articleFormsCount].order}
```

Comment:
refactor as const mainArticle = mainArticles[articleFormsCount]. And then use mainArticle.order... or something like that for not repeating yourself

Reply:
And consider making props the same name as here to do it all with unpacking syntax {...mainArticle}. This will reduce the amount of unneeded lines